### ISSUE TITLE
False positives for unused pragmas with DerivingStrategies

### ISSUE THREAD

**Issue by BlackBrane** [ndmitchell/hlint#478]
Both GeneralizedNewtypeDeriving and DeriveAnyClass may incorrectly be reported as unused when they are used via DerivingStrategies.

```
{-# LANGUAGE DerivingStrategies         #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}

newtype X = X Int
    deriving newtype Show

```

Yields 

```
Bad.hs:2:1: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE GeneralizedNewtypeDeriving #-}
Why not remove it.
```

Tested with v2.1.4.

It also happens when deriving with StandaloneDeriving.

**Comment by ndmitchell** [comment]
Thanks for the report! A definite bug, should be easy to solve.

**Comment by BlackBrane** [comment]
Cool, thanks for your work on the libraries!

**Comment by ndmitchell** [comment]
I ended up rewriting the deriving vs extension check entirely, which is much cleaner now you can talk about deriving strategies. That should fix all the bugs, but it's a fairly subtle area, so let me know if you find anything else. Should be released tomorrow.
